### PR TITLE
Refactor/types maintainability

### PR DESCRIPTION
**Pull request by p4535992**
updated the typescript library where possible, the hope is to save a couple of hours of time for those who want to pick up the project again.
Hi, glad to see that you are still interested in starting the project, I intended to try it myself, but unfortunately it is too "smart" for me, I would need a little more time to analyze it.

### CODE REVIEW

**Comment by misterpotts** [comment]
I've got time set aside to finish the changes for Foundry v7 later this week. Then, from there, perhaps we can bring in some of these changes as separate, smaller PR's? 

For example, I'm a fan of the issue templates, and see that as being an easy, zero conflict change to get reviewed and merged. Same might be true of the workflow (release) changes, though I'd like to discuss the intent of the changes a bit. 

However, readme here is behind main so we can perhaps ignore that, I'm not sure about the value of some of the proposed changes in LICENSE, adding linting rules probably requires we agree on linting rules first and any removal of tests is going to be a hard PR to accept. Bundling all that in with the easier stuff makes getting both rolled into the project harder.

Basically, I really appreciate your contribution(s) - I do, thank you 🙏🏻 However they'd be much easier to review and accept if they were smaller, more isolated changes we could discuss more easily. 

I might try splitting some of this out later in the week to get the ball rolling.

**Comment by p4535992** [comment]
as you prefer, I follow you, use any  linting rule you want. I was just updating to version 9 of the typescript and applying the basis of the multisystem design pattern used by [Item Piles](https://github.com/fantasycalendar/FoundryVTT-ItemPiles) which turns out to be particularly robust and dynamic, for now I have no development plan I am just preparing the basis for anyone who wants it put and hands on.
I 'll close this PR and wait for see the final v7 release like a starting point.
